### ISSUE TITLE
whitelisting is very unclear

### ISSUE THREAD

**Issue by hyperbolic2346** [itzg/minecraft-server-charts#20]
The values.yaml says to have a comma separated list of usernames, but the docker container seems to expect true/false with whiltelist in whitelist.json. The chart implies it will turn on whitelisting and generate this json file, but that doesn't seem to be the case. I would suggest either to update the docs to indicate how to turn on whitelisting and mention the json file needs to be filled out or to make the chart write the whitelist.json on start with an init container or the like.

**Comment by itzg** [comment]
The whitelist variable itself is referring to https://github.com/itzg/docker-minecraft-server#whitelist-players , which does end up with the json file getting updated at server startup. With that said, I haven't double checked that it works lately, but I do know the `OPS` setting works correctly in the same way.

**Comment by hyperbolic2346** [comment]
Something is wrong then. When I set `whitelist: player1,player2` I get this:

```
$ kubectl -n minecraft logs minecraft-bedrock-minecraft-bedrock-9ff8555d6-s7swv
DEBU[0000] Using /data to match uid and gid
DEBU[0000] Resolved UID=0 from match path
DEBU[0000] Resolved GID=0 from match path
2020/10/25 14:38:19 Failed to bulk-set properties in file: Value 'player1,player2' for property white-list is in allowed list: [true false]
```

**Comment by hyperbolic2346** [comment]
Oh, I just noticed you linked minecraft and I am using bedrock. I wonder if that is the disconnect here. That appears to pass directly through to the server.

**Comment by itzg** [comment]
Ah, I wonder if the contributor of the bedrock chart accidentally copied that option over as-is. I see what you mean now that it is supposed to be a boolean option in bedrock.

**Comment by hyperbolic2346** [comment]
Yeah, I'm working on a PR to port over the java edition code to build the whitelist.

**Comment by hyperbolic2346** [comment]
Opened https://github.com/itzg/docker-minecraft-bedrock-server/pull/101 to make this behave as it is described in the values.yaml
